bin/seattle/pull-prod-admin-config

```diff
@@ -0,0 +1,147 @@
+    --query 'Subnets[0].SubnetId' \
+    --output text) && echo "Found public subnet (ID: $SUBNET_ID) to launch the instance into"
+
+# Launch a small ubuntu VM
+INSTANCE_ID=$(aws ec2 run-instances \
+    --image-id ami-03d5c68bab01f3496 \
+    --count 1 \
+    --instance-type t2.micro \
+    --key-name $KEY_NAME \
+    --security-group-ids $SG_ID \
+    --subnet-id $SUBNET_ID \
+    --query 'Instances[0].InstanceId' \
+    --output text) && \
+    echo "Created EC2 instance (ID: $INSTANCE_ID) and waiting for it to be ready. This may take 3-5 minutes."
+aws ec2 create-tags --resources $INSTANCE_ID \
+    --tags Key=Name,Value=MyEmergencyDBAccessInstance$TIMESTAMP$ENVIRONMENT
+
+# Wait until the instance is available for connection
+until aws ec2 wait instance-status-ok --instance-ids $INSTANCE_ID; do :; done
+INSTANCE_IPADDRESS=$(aws ec2 describe-instances \
+    --instance-ids $INSTANCE_ID \
+    --query 'Reservations[0].Instances[0].PublicIpAddress' \
+    --output text) && echo "Instance public IP: $INSTANCE_IPADDRESS"
+
+# Find the DB address
+DB_ADDRESS=$(aws rds describe-db-instances \
+    --db-instance-identifier $ENVIRONMENT-civiform-postgres-db \
+    --query 'DBInstances[0].Endpoint.Address' \
+    --output text) && echo "DB address: $DB_ADDRESS"
+
+# Get the DB password
+DB_PASSWORD=$(aws secretsmanager get-secret-value \
+    --secret-id $DB_PASSWORD_ID \
+    --query 'SecretString' \
+    --output text)
+
+# Connect to the instance, install psql, and connect to DB
+echo "Connecting to the database"
+ssh -q -o UserKnownHostsFile=/dev/null \
+    -o StrictHostKeyChecking=no \
+    -o IdentitiesOnly=yes \
+    -t -i $KEY_FILE \
+    ubuntu@$INSTANCE_IPADDRESS \
+    "yes | sudo apt-get update > /dev/null; \
+        yes | sudo apt-get install postgresql-client > /dev/null; \
+        PGPASSWORD='$DB_PASSWORD' pg_dump -w -Fc \
+        -h $DB_ADDRESS \
+        -d postgres \
+        -U civiform \
+        -t programs \
+        -t questions \
+        -t versions \
+        -t versions_programs \
+        -t versions_questions \
+        postgres > program_data.dump"
+
+echo "Copying file to local computer using scp"
+scp -o UserKnownHostsFile=/dev/null \
+  -o StrictHostKeyChecking=no \
+  -o IdentitiesOnly=yes \
+  -i $KEY_FILE ubuntu@${INSTANCE_IPADDRESS}:/home/ubuntu/program_data.dump .
+
+# Terminate the instance
+echo "Terminating the instance"
+aws ec2 terminate-instances --instance-ids $INSTANCE_ID
+
+# Delete the emergency key pair
+echo "Removing key pair"
+aws ec2 delete-key-pair --key-name $KEY_NAME
+rm -f $KEY_FILE
+
+# Remove the emergency security group from DB security group
+echo "Revoke access to DB security group"
+aws ec2 revoke-security-group-ingress \
+    --group-id $DBSG_ID \
+    --protocol tcp \
+    --port 0-65535 \
+    --source-group $SG_ID
+
+# Remove the emergency security group
+echo "Removing the emergency security group"
+until aws ec2 wait instance-terminated --instance-ids $INSTANCE_ID; do :; done
```

Comment:
Is the : in "do :" correct?  What's it doing?

should there be a sleep in here?

Reply:
uhh good question

Reply:
i have no idea why it would be done this way vs

```
aws ec2 wait instance-terminated --instance-ids $INSTANCE_ID
```